### PR TITLE
fixes #1643, not display fb,tw,email buttons if not configured

### LINKED ISSUE #1643: Do not display FB, Email and Twitter buttons if not configured in Snippet
For petition pages, please do not display the FB, Email and Tweet buttons if the snippet related to the petition doesn't have these options configured

### PR DESCRIPTION
**Pull request by techac**
Closes #1643 

[ ] Generates new fake data, if necessary
[ ] Wagtail documentation has been updated, if necessary
[ ] Includes tests, if appropriate
[ ] Migrations squashed

### CODE REVIEW

**Comment by techac** [comment]
@alanmoo you may review the PR

**Comment by alanmoo** [comment]
Thanks- I'll try to get to it today, though we're trying to tie up the Buyer's Guide so our petition work is a little bit of a lower priority this week.

**Comment by techac** [comment]
@alanmoo thanks

**Comment by techac** [comment]
@alanmoo can you please update me with my PR

File: source/js/components/petition/petition.jsx [hunk 1]
```diff
@@ -443,24 +443,29 @@ export default class Petition extends React.Component {
...
-              </svg>
-              TWEET
-            </button>
-            <button className="share-progress-btn share-progress-em" onClick={e => this.emButtonClicked(e)}>
-              <svg aria-hidden="true" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512">
-                <path fill="currentColor" d="M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z"></path>
-              </svg>
-              EMAIL
-            </button>
+            {this.props.shareLink ?
```
Comment (alanmoo) [inline source/js/components/petition/petition.jsx]: This code will never execute. The block on line 435 comes after a check for the same variable, so if `this.props.shareLink` exists, line 435 will be returned and skip the remainder of this function.

There should be multiple checks for variables to resolve this issue- one for facebook, one for twitter, and one for email. The variable you're looking for is the ones defined [here](https://github.com/mozilla/foundation.mozilla.org/blob/master/network-api/networkapi/wagtailpages/models.py#L300-L306). You're going to have to inspect the code to figure out how they eventually get passed to the client.

Comment (techac) [reply source/js/components/petition/petition.jsx]: @alanmoo  I got it.. and when I inspected the code there was no prop for sharelink_twiiter, facebook and email ... can you plz help me how can I get these fields from models to jsx as I am new to react... thanks ...

Comment (alanmoo) [reply source/js/components/petition/petition.jsx]: Unfortunately if I had the time to trace that, I would be solving this ticket myself. If you can't figure it out, I'd suggest checking out our [good first issues](https://github.com/mozilla/foundation.mozilla.org/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22)

**Comment by Pomax** [comment]
I'm going to close this for now, as it's been unresolved for a few months - once this issue makes its way back into the project board, we can revisit it.

**Pull request closed without merging**